### PR TITLE
NH-65069 APM Python lambda layer build on local

### PR DESCRIPTION
**Pull request by tammy-baylis-swi**
Adds APM Python lambda layer creation on local with `make aws-lambda` (while in a manylinux container via `docker-compose run aarch64` or `x86_64`).

Main difference between this and the hack in the lambda PR (https://github.com/solarwinds/opentelemetry-lambda/pull/1) is this approach uses local Python code, instead of installing from TestPyPI (much slower when a new APM has to be published).

I like how this repurposes an AO rule vestige, uses the Makefile, and is relatively simple.

What I'm not happy with is how this has duplicates of `otel-instrument` and `otel_wrapper.py` scripts from this part of the lambda repo: https://github.com/solarwinds/opentelemetry-lambda/tree/main/python/src/otel/otel_sdk. Also the resulting zip file is 9.5MB right now. 🙃 

Compatible with Python 3.7 and 3.8 for now. Will add 3.9, 3.10, maybe 3.11 later and test all of them.

Tested in a lambda environment with aarch64 and Python 3.8 for now with
1. temporarily merging this branch into the [settings API integration PoC branch](https://github.com/solarwindscloud/solarwinds-apm-python/pull/207)
2. using `make aws-lambda`
3. uploading the resulting `dist/solarwinds_apm_python.zip` to a lambda layer [tammy_test_apm_python_m1_buildtest](https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/layers/tammy_test_apm_python_m1_buildtest)
4. adding layer to a function equipped with otelcol, [tammy_test_python_m1_buildtest](https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/tammy_test_python_m1_buildtest?tab=monitoring) 
5. invoking function to get a 200
6. and checking Cloudwatch ([example logs](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:log-groups/log-group/%2Faws%2Flambda%2Ftammy_test_python_m1_buildtest/log-events/2023%2F11%2F01%2F[%24LATEST]71bddc8d4d9944acb2a257a9cf320172?start=PT3H))
7. and checking traces exporter via otelcol to SWO NA-02!: https://my.na-02.cloud.solarwinds.com/158363224169902080/traces/56E544951FF126E3DDDCADBF52E19F9F/2F68E46BF9F7AE45/details/breakdown?perspective=requests&serviceId=e-1656498477873647616&sort_by_breakdown=spanLayer&sort_order_breakdown=ascending

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -179,27 +185,36 @@ package: sdist check-sdist-local manylinux-wheels check-wheel-local
...
+	@/opt/python/cp38-cp38/bin/pip3.8 install . -t ${target_dir}/nodeps --no-deps
 	@echo -e "Removing non-lambda C-extension library files generated by pip install under target directory."
-	@rm ${target_dir}/solarwinds_apm/extension/*.so*
+	@rm ${target_dir}/nodeps/solarwinds_apm/extension/*.so*
 	@echo -e "Building AWS Lambda version of C-extensions for all supported Python versions in target directory."
-	@set -e; for PYBIN in cp37-cp37m cp38-cp38; do /opt/python/$${PYBIN}/bin/python setup.py build_ext_lambda -b ${target_dir}; done
+	@set -e; for PYBIN in cp37-cp37m cp38-cp38; do /opt/python/$${PYBIN}/bin/python setup.py build_ext_lambda -b ${target_dir}/nodeps; done
 	@echo -e "Copying AWS Lambda specific Oboe library liboboe-1.0-lambda-${platform}.so into target directory."
-	@cp solarwinds_apm/extension/liboboe-1.0-lambda-${platform}.so ${target_dir}/solarwinds_apm/extension/liboboe.so
-	@rm -rf ${target_dir}/*-info
```
Comment (tammy-baylis-swi) [inline Makefile:199]: App running in lambda cannot find the above installed modules without the `*.dist-info` contents.

File: Makefile [hunk 2]
```diff
@@ -179,27 +185,36 @@ package: sdist check-sdist-local manylinux-wheels check-wheel-local
...
 	@echo -e "Creating target directory ${target_dir} for AWS Lambda layer artifacts."
-	mkdir -p ${target_dir}
+	mkdir -p ${target_dir}/python
+	@echo -e "Install upstream dependencies to include in layer"
+	@/opt/python/cp38-cp38/bin/pip3.8 install -t ${target_dir}/python -r lambda/requirements.txt
+	@echo -e "Install upstream dependencies without deps to include in layer"
+	@/opt/python/cp38-cp38/bin/pip3.8 install -t ${target_dir}/nodeps -r lambda/requirements-nodeps.txt --no-deps
 	@echo -e "Install solarwinds_apm to be packed up in zip archive to target directory."
-	@/opt/python/cp38-cp38/bin/pip3.8 install . -t ${target_dir}
+	@/opt/python/cp38-cp38/bin/pip3.8 install . -t ${target_dir}/nodeps --no-deps
```
Comment (tammy-baylis-swi) [inline Makefile:202]: We install `solarwinds_apm` and some deps with `--no-deps` and in a separate folder else not all packages will be fully downloaded and accessible at `opentelemetry/instrumentation` path ([this comment](https://github.com/solarwinds/opentelemetry-lambda/blob/main/python/src/otel/Dockerfile#L12-L13)).

**Comment by xuan-cao-swi** [comment]
Not sure if this will help: https://github.com/signalfx/splunk-otel-lambda/blob/main/python/build-layer.sh#L45

**Review by xuan-cao-swi — APPROVED** [review]
LGTM

**Comment by tammy-baylis-swi** [comment]
> Not sure if this will help: https://github.com/signalfx/splunk-otel-lambda/blob/main/python/build-layer.sh#L45

Thanks @xuan-cao-swi ! It'll be worth a try in my next iteration. Same with this similar cleanup: https://github.com/solarwinds/opentelemetry-lambda/blob/main/python/src/otel/Dockerfile#L19-L20

**Pull request merged**